test/e2e/storage/pd.go

```diff
@@ -451,29 +453,34 @@ func countReadyNodes(c clientset.Interface, hostName types.NodeName) int {
 }
 
 func verifyPDContentsViaContainer(f *framework.Framework, podName, containerName string, fileAndContentToVerify map[string]string) {
+	var err error
+	var v string
+	errs := []error{}
 	for filePath, expectedContents := range fileAndContentToVerify {
-		var value string
-		// Add a retry to avoid temporal failure in reading the content
-		for i := 0; i < maxReadRetry; i++ {
-			v, err := f.ReadFileViaContainer(podName, containerName, filePath)
-			value = v
+		// Add a timed retry to avoid temporal failure in reading the content
+		success := false
+		err = nil
+		for start := time.Now(); time.Since(start) < time.Minute; time.Sleep(10 * time.Second) {
+			v, err = f.ReadFileViaContainer(podName, containerName, filePath)
 			if err != nil {
-				framework.Logf("Error reading file: %v", err)
+				err = fmt.Errorf("error reading file via container: %v", err)
+				continue
 			}
-			framework.ExpectNoError(err)
-			framework.Logf("Read file %q with content: %v (iteration %d)", filePath, v, i)
+			framework.Logf("Read file %q with content: %v (time since first iteration %v)", filePath, v, time.Since(start))
 			if strings.TrimSpace(v) != strings.TrimSpace(expectedContents) {
-				framework.Logf("Warning: read content <%q> does not match execpted content <%q>.", v, expectedContents)
-				size, err := f.CheckFileSizeViaContainer(podName, containerName, filePath)
-				if err != nil {
-					framework.Logf("Error checking file size: %v", err)
-				}
-				framework.Logf("Check file %q size: %q", filePath, size)
-			} else {
-				break
+				err = fmt.Errorf("read content <%q> does not match expected content <%q>", v, expectedContents)
+				continue
 			}
+			// All checks completed - Success
+			success = true
+			break
+		}
+		if !success {
+			errs = append(errs, fmt.Errorf("could not verify that contents of file %v : %v", filePath, err))
 		}
-		framework.ExpectEqual(strings.TrimSpace(value), strings.TrimSpace(expectedContents))
```

Comment:
The above check was redundant and therefore being completely removed, correct?

Reply:
yes